### ISSUE TITLE
stucked creating the volume

### ISSUE THREAD

**Issue by cornelius-keller** [quobyte/quobyte-deploy#6]
Hi @quofelix @sttts,

I tried the kubernets-complete example. First issue was that I had to change the mountpoint of the host directory form /devices to /devices/dev1 in both:
https://github.com/quobyte/quobyte-deploy/blob/master/kubernetes/quobyte-complete/registry-bootstrap-pod.yaml#L48 
https://github.com/quobyte/quobyte-deploy/blob/master/kubernetes/quobyte-complete/registry-rc.yaml#L50

Otherwise it seems that /devices/dev1 is not recognized as  mount point and the start of the registry pod fails.

After solving this I am now finally stucked at https://github.com/quobyte/quobyte-deploy/tree/master/kubernetes/quobyte-complete#create-a-volume .

The result I am getting is:

```
kubectl exec -it --namespace=quobyte registry-bootstrap -- qmgmt -u api:7860 volume create cluster root root BASE 0777
Operation denied by API service because: 'NOT_ENOUGH_RESOURCES/POSIX_ERROR_NONE: Could not create volume cluster because available metadata device do not satisfy the requested replica configuration: 1, got distinct devices [] out of list[] available devices. / ' Code: -20
error: error executing remote command: Error executing command in container: Error executing in Docker Container: 249
```

It seems that the metadata pod is not picking up the mountpoints of the other pods in my case. Logs of the metadata pods all show:

```
[ W | a                    | deviceStatus... | 118 | May 24 17:17:19.792 ] there are currently no data devices available
[ W | a                    | deviceStatus... | 118 | May 24 17:17:29.798 ] there are currently no data devices available
[ W | a                    | deviceStatus... | 118 | May 24 17:17:39.803 ] there are currently no data devices available
[ W | a                    | deviceStatus... | 118 | May 24 17:17:49.809 ] there are currently no data devices available
[ W | a                    | deviceStatus... | 118 | May 24 17:17:59.814 ] there are currently no data devices available
[ W | a                    | deviceStatus... | 118 | May 24 17:18:01.199 ] there are currently no data devices available
[ W | a                    | deviceStatus... | 118 | May 24 17:18:11.295 ] there are currently no data devices available
```

I did moify the MountFlags of the docker daemon as descibed in:
https://github.com/quobyte/quobyte-deploy/blob/master/kubernetes/Quobyte_readme.md#mountflags

and verified that it worked on all the nodes by executing: 

```
core@core06 ~ $ systemctl show docker.service | grep MountFlags=
MountFlags=1048576
```

The devices are marked as online:

```
kubectl exec -it --namespace=quobyte registry-bootstrap -- qmgmt -u api:7860 device list
Id Host                                     Status                Disk Used Disk Avail  Services  
3 registry-6pclc                           ONLINE                    17 GB     240 GB  REGISTRY  
1 registry-bootstrap                       ONLINE                    19 GB     240 GB  REGISTRY  
2 registry-fsoyh                           ONLINE                    17 GB     246 GB  REGISTRY  
4 registry-hzlq6                           ONLINE                    17 GB     240 GB  REGISTRY  
```

Coreos Version is: `CoreOS stable (899.17.0)`

As I said I am stucked. Is there anything I am doing obviously wrong? 
As far as I can tell it seems that the mounts of the registry containers are not seen in the metadata container. But I don't know why and what I need to do to change this.

If i execute a mount inside the metadata container I see only its own mounts even if there is a registry container running on the same node:

```
core@core02 ~ $ # get all quobyte containers
core@core02 ~ $ docker ps | grep quobyte
eb8e0f758bd8        quay.io/quobyte/quobyte-server:1.2.10            "/bin/bash -xec 'sed "   37 minutes ago      Up 37 minutes                                  k8s_quobyte-metadata.e9a3b333_metadata-y7sgl_quobyte_4edf595b-21d3-11e6-a679-901b0e93232b_3cdbcb2a
4d39049cec4f        gcr.io/google_containers/pause:2.0               "/pause"                 37 minutes ago      Up 37 minutes                                  k8s_POD.6059dfa2_metadata-y7sgl_quobyte_4edf595b-21d3-11e6-a679-901b0e93232b_9721f108
c66c7b66970c        quay.io/quobyte/quobyte-server:1.2.10            "/bin/bash -xec 'sed "   37 minutes ago      Up 37 minutes                                  k8s_quobyte-metadata.56a145f2_data-jz7gf_quobyte_4953caea-21d3-11e6-a679-901b0e93232b_f3edbbf6
f4bca29a10df        gcr.io/google_containers/pause:2.0               "/pause"                 37 minutes ago      Up 37 minutes                                  k8s_POD.6059dfa2_data-jz7gf_quobyte_4953caea-21d3-11e6-a679-901b0e93232b_0722a6e4
bd923e2a9000        quay.io/quobyte/quobyte-server:1.2.10            "/bin/bash -xec 'sed "   48 minutes ago      Up 48 minutes                                  k8s_quobyte-registry.36244138_registry-6pclc_quobyte_d65715de-21d1-11e6-a679-901b0e93232b_bfe9b78b
e688eb3cd6a6        gcr.io/google_containers/pause:2.0               "/pause"                 48 minutes ago      Up 48 minutes                                  k8s_POD.d8224cbf_registry-6pclc_quobyte_d65715de-21d1-11e6-a679-901b0e93232b_35e265d7
core@core02 ~ $ # execute mount in registry container
core@core02 ~ $ docker exec -ti bd923e2a9000 mount
overlay on / type overlay (rw,relatime,lowerdir=/var/lib/docker/overlay/38c2b075d4ec64afbcf01a7fbb1f31949297a23aac0de03db3825300575aa50d/root,upperdir=/var/lib/docker/overlay/bd923e2a90007066d7a284c957d0e01c67d2da9a9812f9566d6f15481d432109/upper,workdir=/var/lib/docker/overlay/bd923e2a90007066d7a284c957d0e01c67d2da9a9812f9566d6f15481d432109/work)
proc on /proc type proc (rw,nosuid,nodev,noexec,relatime)
tmpfs on /dev type tmpfs (rw,nosuid,seclabel,mode=755)
devpts on /dev/pts type devpts (rw,nosuid,noexec,relatime,seclabel,gid=5,mode=620,ptmxmode=666)
sysfs on /sys type sysfs (ro,nosuid,nodev,noexec,relatime,seclabel)
tmpfs on /sys/fs/cgroup type tmpfs (rw,nosuid,nodev,noexec,relatime,seclabel,mode=755)
cgroup on /sys/fs/cgroup/systemd type cgroup (rw,nosuid,nodev,noexec,relatime,xattr,release_agent=/usr/lib/systemd/systemd-cgroups-agent,name=systemd)
cgroup on /sys/fs/cgroup/freezer type cgroup (rw,nosuid,nodev,noexec,relatime,freezer)
cgroup on /sys/fs/cgroup/net_cls,net_prio type cgroup (rw,nosuid,nodev,noexec,relatime,net_cls,net_prio)
cgroup on /sys/fs/cgroup/perf_event type cgroup (rw,nosuid,nodev,noexec,relatime,perf_event)
cgroup on /sys/fs/cgroup/memory type cgroup (rw,nosuid,nodev,noexec,relatime,memory)
cgroup on /sys/fs/cgroup/blkio type cgroup (rw,nosuid,nodev,noexec,relatime,blkio)
cgroup on /sys/fs/cgroup/cpu,cpuacct type cgroup (rw,nosuid,nodev,noexec,relatime,cpu,cpuacct)
cgroup on /sys/fs/cgroup/cpuset type cgroup (rw,nosuid,nodev,noexec,relatime,cpuset)
cgroup on /sys/fs/cgroup/devices type cgroup (rw,nosuid,nodev,noexec,relatime,devices)
/dev/sda9 on /devices/dev1 type ext4 (rw,relatime,seclabel,data=ordered)
/dev/sda9 on /dev/termination-log type ext4 (rw,relatime,seclabel,data=ordered)
tmpfs on /run/secrets/kubernetes.io/serviceaccount type tmpfs (ro,relatime,rootcontext=system_u:object_r:var_lib_t:s0,seclabel)
/dev/sda9 on /etc/resolv.conf type ext4 (rw,relatime,seclabel,data=ordered)
/dev/sda9 on /etc/hostname type ext4 (rw,relatime,seclabel,data=ordered)
/dev/sda9 on /etc/hosts type ext4 (rw,relatime,seclabel,data=ordered)
shm on /dev/shm type tmpfs (rw,nosuid,nodev,noexec,relatime,context="system_u:object_r:svirt_lxc_file_t:s0:c253,c520",size=65536k)
mqueue on /dev/mqueue type mqueue (rw,nosuid,nodev,noexec,relatime,seclabel)
core@core02 ~ $ # execute mount in metadata container
core@core02 ~ $ docker exec -ti eb8e0f758bd8 mount
overlay on / type overlay (rw,relatime,lowerdir=/var/lib/docker/overlay/38c2b075d4ec64afbcf01a7fbb1f31949297a23aac0de03db3825300575aa50d/root,upperdir=/var/lib/docker/overlay/eb8e0f758bd8923c2574e11976d82b6156940ea1865a775d7db23375c0a90956/upper,workdir=/var/lib/docker/overlay/eb8e0f758bd8923c2574e11976d82b6156940ea1865a775d7db23375c0a90956/work)
proc on /proc type proc (rw,nosuid,nodev,noexec,relatime)
tmpfs on /dev type tmpfs (rw,nosuid,seclabel,mode=755)
devpts on /dev/pts type devpts (rw,nosuid,noexec,relatime,seclabel,gid=5,mode=620,ptmxmode=666)
sysfs on /sys type sysfs (ro,nosuid,nodev,noexec,relatime,seclabel)
tmpfs on /sys/fs/cgroup type tmpfs (rw,nosuid,nodev,noexec,relatime,seclabel,mode=755)
cgroup on /sys/fs/cgroup/systemd type cgroup (rw,nosuid,nodev,noexec,relatime,xattr,release_agent=/usr/lib/systemd/systemd-cgroups-agent,name=systemd)
cgroup on /sys/fs/cgroup/freezer type cgroup (rw,nosuid,nodev,noexec,relatime,freezer)
cgroup on /sys/fs/cgroup/net_cls,net_prio type cgroup (rw,nosuid,nodev,noexec,relatime,net_cls,net_prio)
cgroup on /sys/fs/cgroup/perf_event type cgroup (rw,nosuid,nodev,noexec,relatime,perf_event)
cgroup on /sys/fs/cgroup/memory type cgroup (rw,nosuid,nodev,noexec,relatime,memory)
cgroup on /sys/fs/cgroup/blkio type cgroup (rw,nosuid,nodev,noexec,relatime,blkio)
cgroup on /sys/fs/cgroup/cpu,cpuacct type cgroup (rw,nosuid,nodev,noexec,relatime,cpu,cpuacct)
cgroup on /sys/fs/cgroup/cpuset type cgroup (rw,nosuid,nodev,noexec,relatime,cpuset)
cgroup on /sys/fs/cgroup/devices type cgroup (rw,nosuid,nodev,noexec,relatime,devices)
/dev/sda9 on /devices type ext4 (rw,relatime,seclabel,data=ordered)
/dev/sda9 on /dev/termination-log type ext4 (rw,relatime,seclabel,data=ordered)
tmpfs on /run/secrets/kubernetes.io/serviceaccount type tmpfs (ro,relatime,rootcontext=system_u:object_r:var_lib_t:s0,seclabel)
/dev/sda9 on /etc/resolv.conf type ext4 (rw,relatime,seclabel,data=ordered)
/dev/sda9 on /etc/hostname type ext4 (rw,relatime,seclabel,data=ordered)
/dev/sda9 on /etc/hosts type ext4 (rw,relatime,seclabel,data=ordered)
shm on /dev/shm type tmpfs (rw,nosuid,nodev,noexec,relatime,context="system_u:object_r:svirt_lxc_file_t:s0:c691,c956",size=65536k)
mqueue on /dev/mqueue type mqueue (rw,nosuid,nodev,noexec,relatime,seclabel)
core@core02 ~ $ # see if docker has MountFlags=shared
core@core02 ~ $ systemctl show docker.service | grep MountFlags=
MountFlags=1048576
```

Thanks in advance for helping
Cornelius

**Comment by quofelix** [comment]
You need at least one data and one metadata device and service.

**Comment by cornelius-keller** [comment]
Sorry for the confusion about mounts not beeing visible in other containers. I got this totally wrong and should have thought deeper about it.
The reason for this not working properly is that quobyte changed the way it scans for volume markers. The volume marker needs do be of the mount point, in a subdirectory it is not working.

Thanks to Silvan at quobyte for helping me out and clarifying things.

The only problem left for me is that the client-nsenter pod was not working for me due to shared library problems and the workaround I used without nsenter is running but mounting the quebyte cluster storage only in the container and not in the host system.
